### PR TITLE
[feature]: Display "no payment method available" message if no allowed or configured payment methods are present

### LINKED ISSUE #2854: [bug]: It's not showing any message when there are no supported payment methods available
**Describe the bug**
It's not showing message when there are no supported payment methods available.
It's becoming and empty container instead of showing proper message.

**To reproduce**
1. Make sure you have braintree payment method disabled and other payment methods like Purchase order, bank transfer enabled.
2. Add any product to cart
3. Go to Cart -> Checkout
4. Add shipping address and select any shipping method
5. Let it load the payment methods from server. After loading completion payment method container will become empty as shown in below screenshot.

**Expected behavior**
It should show message like "No Payment method available.", but it's not showing anything and container becomes empty as shown in below screenshot.
Screen Shot 2020-11-10 at 4 42 19 PM

**Screenshots**
Screen Shot 2020-11-10 at 4 22 34 PM
Screen Shot 2020-11-10 at 4 22 59 PM

**Additional context**
Make sure you have braintree payment method disabled and other payment methods like Purchase order, bank transfer enabled.

**Possible solutions**
Adding message will do the job

**Please complete the following device information:**
 - Device [e.g. iPhone6, PC, Mac, Pixel3]:
 - OS [e.g. iOS8.1, Windows 10]:
 - Browser [e.g. Chrome, Safari]:
 - Browser Version [e.g. 22]:
 - Magento Version:
 - PWA Studio Version:
 - NPM version `npm -v`:
 - Node Version `node -v`:

**Please let us know what packages this bug is in regards to:**
- [ ] `venia-concept`
- [x] `venia-ui`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by ankurvr**
## Description

It's not showing message when there are no supported payment methods available.
It's becoming and empty container instead of showing proper message.

## Related Issue

Closes #2854.
[PWA-1115](https://jira.corp.magento.com/browse/PWA-1115)
### Verification Steps

1. Make sure you have braintree payment method disabled and other payment methods like Purchase order, bank transfer enabled.
2. Add any product to cart
3. Go to Cart -> Checkout
4. Add shipping address and select any shipping method
5. Let it load the payment methods from server. After loading completion payment method it should show the message.

## Screenshots / Screen Captures (if appropriate)
[Image from Gyazo](https://gyazo.com/ff4cba4f3bb6f739c187eb7edc47d451)

[Image from Gyazo](https://gyazo.com/a8b9de00d88079b2c38fbc6a1e018fc3)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js [hunk 1]
```diff
@@ -36,37 +36,48 @@ const PaymentMethods = props => {
...
-    const radios = availablePaymentMethods.map(({ code, title }) => {
-        // If we don't have an implementation for a method type, ignore it.
-        if (!Object.keys(payments).includes(code)) {
-            return;
-        }
+    const radios = availablePaymentMethods
+        .map(({ code, title }) => {
+            // If we don't have an implementation for a method type, ignore it.
+            if (
+                !Object.keys(PAYMENT_METHOD_COMPONENTS_BY_CODE).includes(code)
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js:43]: Where does `PAYMENT_METHOD_COMPONENTS_BY_CODE` come from?

Comment (ankurvr) [reply packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js]: Sorry i copied it from the released package. I am reverting it back

**Review by ankurvr — COMMENTED** [review]
The requested change is done.

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js [hunk 2]
```diff
@@ -36,37 +36,48 @@ const PaymentMethods = props => {
...
+                </div>
+            );
+        })
+        .filter(paymentMethod => !!paymentMethod);
+
+    const noPaymentMethodMessage = !radios.length ? (
+        <span className={classes.no_payment_method}>
+            {formatMessage({
+                id: 'checkoutPage.noPaymentMethodAvailable',
+                defaultMessage: 'No Payment method available.'
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js:77]:
```suggestion
                defaultMessage: 'No payment method available.'
```

File: packages/venia-ui/i18n/en_US.json [hunk 3]
```diff
@@ -74,6 +74,7 @@
     "checkoutPage.paymentInformationStep": "3. Payment Information",
     "checkoutPage.paymentMethodStatus": "{selectedPaymentMethod} is not supported for editing.",
     "checkoutPage.paymentSummary": "{cardType} ending in {lastFour}",
+    "checkoutPage.noPaymentMethodAvailable": "No Payment method available.",
```
Comment (sirugh) [inline packages/venia-ui/i18n/en_US.json:77]:
```suggestion
    "checkoutPage.noPaymentMethodAvailable": "No payment method available.",
```

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js [hunk 4]
```diff
@@ -36,37 +36,48 @@ const PaymentMethods = props => {
...
+                    />
+                    {renderedComponent}
+                </div>
+            );
+        })
+        .filter(paymentMethod => !!paymentMethod);
+
+    const noPaymentMethodMessage = !radios.length ? (
+        <span className={classes.no_payment_method}>
+            {formatMessage({
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js:75]: You'll need to make sure this method is available.

```es6
import { useIntl } from 'react-intl';

....

const { formatMessage } = useIntl();
```

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.css [hunk 5]
```diff
@@ -15,3 +15,7 @@
     font-weight: 600;
     justify-self: start;
 }
+
+.no_payment_method {
+    padding: 1.5rem;
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.css:20]: To get this component more aligned with the others (shipping address/method) lets just add `padding: 2rem;` to the `root` class above, and remove this class. Also, remove the padding from the `payment_method` class.

**Review by sirugh — CHANGES_REQUESTED** [review]

**Comment by sirugh** [comment]
@ankurvr I opened an internal tracking ticket for the team to prioritize. You may not see movement on it for a bit, but I've given you some feedback that I think will capture most of the concerns with the PR. Please make changes when you can and we will take it from there.

**Comment by ankurvr** [comment]
@sirugh i will take care of the requested changes. Thanks

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js (hunk 4 above)
Comment (ankurvr) [reply packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js:78]: Added.

**Review by ankurvr — COMMENTED** [review]
Fixed suggested changes in CSS

**Review by sirugh — APPROVED** [review]

**Comment by schensley** [comment]
@ankurvr I am going off the screen shots provided here. I am concerned that the treatment is somewhat ambiguous, in that it looks like everything is responding as it should, but payments being 'offline' is actually a problem and to communicate that to the shopper I think it should look like an error/alert message.  Additionally, I can't tell if the button to continue is active or what they shopper can do here. If there is an active button and tapping it then throws an error message that there is no payment, I think that would be aggravating for the shopper. Happy to provide a mock-up of you'd like.

**Comment by ankurvr** [comment]
@schensley Sorry i am bit confused with your comment.
Please correct me if i am wrong.
As per screenshot you want me to change the suggested functionality. Like message in red color in payments box and showing error message at top?

**Comment by schensley** [comment]
@ankurvr Sorry, I was trying to communicate the rationale behind the request for adding this "error" style messaging. 
Yes, ideally a 2-part system of messaging, where there is a "headline" error message at the top, and a more specific (if possible) error message in the location of problem. The salient point being that the problem is highlighted for the shopper.

**Comment by sirugh** [comment]
@schensley unfortunately the checkout page does not have a page level error so that will be extra scope.

@ankurvr can you please just update the text to have `color: rgb(var(--venia-global-color-error));` in the error case? If you want to tackle adding a page level error I will be happy to provide some guidance. Let me know.

**Comment by ankurvr** [comment]
@sirugh I have just updated the text message with error color you suggested.
It will look like below screenshot.
pwa-docker localhost_8080_checkout


Also i will be happy to tackle adding page level error. Please guide me to achieve the same.

**Review by sirugh — APPROVED** [review]
Thanks for your work on this! We still need to get that page level error displayed, and disable the review button, but I've scoped that to a different internal ticket. If you'd like to work on that as well, let me know and I can open a github issue with relevant info.

Edit: I just saw you said you would be happy to tackle it. Let me discuss with the team and I'll follow up with you later.

**Comment by sirugh** [comment]
@ankurvr I have opened https://github.com/magento/pwa-studio/issues/2869 for you, with details. Let's continue the conversation in there if you have any questions :)

**Comment by schensley** [comment]
@ankurvr Great!  Looks really good. Thank you.

@sirugh I'm not sure the "Review" button should be disabled. If the shopper didn't notice the message or didn't understand it, if they can press the BTN (which shoppers seek) and it returns an error they might then take a closer look to see the issue is. No harm, we did try to tell them up front.

**Comment by ankurvr** [comment]
@sirugh @schensley Please update the final approach in the newly created task/ticket #2869 so i can proceed accordingly.

**Comment by sirugh** [comment]
@ankurvr not sure what you mean, I've added the desired solution and a mockup in the issue. Let's move questions over to that thread instead of adding more here :)

**Comment by dpatil-magento** [comment]
QA Approved.

**Review by supernova-at — APPROVED** [review]

**Pull request merged**